lib/collection/src/shards/replica_set.rs

```diff
@@ -1133,6 +1140,95 @@ impl ShardReplicaSet {
         Ok(())
     }
 
+    pub async fn queue_proxify_local(&self, remote_shard: RemoteShard) -> CollectionResult<()> {
+        let mut local_write = self.local.write().await;
+
+        match &*local_write {
+            // Expected state, continue
+            Some(Local(_)) => {}
+            // Unexpected states, error
+            Some(QueueProxy(_)) => {
+                return Err(CollectionError::service_error(format!(
+                    "Cannot queue proxify local shard {} to peer {} because it is already queue proxified",
+                    self.shard_id, remote_shard.peer_id,
+                )));
+            }
+            Some(ForwardProxy(proxy)) => {
+                return if proxy.remote_shard.peer_id == remote_shard.peer_id {
+                    Ok(())
+                } else {
```

Comment:
- I don't think this is right for `queue_proxify_local`.
- Should we add the same handling for the `Some(QueueProxy(_))` branch? 🤔

Reply:
But this still stands, I think. Seems like a paste-bug.

Reply:
This is wrong indeed. Fixed it.